libs/atqr/domain/src/lib/domain-spec/stories/visitorAdoptsGoal.disl

```diff
@@ -0,0 +1,32 @@
+read_also api.disl
+read_also enums.disl
+read_also ui.disl
+
+domain ATQR
+  purpose "A platform for people to challenge each other to accomplish goals"
+
+context Challenges
+
+  root Goal
+    @goal: string
+    @challenges: Challenge[]
+    @pursuers: returns @challenges.map(c => c.enrollments.map(e => e.player)).flat()
+
+    CreateChallenge?
+    AdoptGoal? :email :goal
+
+    rules:
+      /**
+      * visitor's email must be validated as a player pursuing or not this goal through Goal's @pursuer
+      */
+      when :email not in @pursuer's email
+        createChallenge
+          @email = visitor's :email input
+          @goal = visitor's :goal of choice
+
+      when @pursuer
+        throw "o usuário já persegue esta meta"
+
+    end rules
```

Comment:
Shouldn't these be inside `AdoptGoal?`. Also, this command needs to carry all the information necessary to create a `Challenge`. You could probably add a command `isPursuer?` that simply validate the email is not among pursuers

Reply:
I changed AdoptGoal?'s name to IsPursuer? cause it makes more sense that way, since this command is about checking exactly if the email belongs to a already pursuer of the goal. But for me IsPursuer? only needs the params that help with this checking and the rest is a job for "createChallenge" method call which is called after in this file.

Reply:
You are correct. And it MUST NOT create a challenge in the process of checking if the pursuer is there. That's why I said "add a isPurser command" and not rename :) You need both